src/main/kotlin/me/partlysanestudios/partlysaneskies/events/EventManager.kt

```diff
@@ -41,19 +45,19 @@ object EventManager {
     fun onChatReceivedEvent(event: ClientChatReceivedEvent) {
         if (event.type.toInt() != 0) return
 
-        if (PSSChatEvent(event.message.formattedText, event.message).post()) {
+        if (PSSChatEvent(event.message.formattedText, event.message).apply { if (event.isCanceled) cancel() }.post()) {
             event.isCanceled = true
         }
     }
 
     @PSSEvent.Subscribe
     fun onChat(event: PSSChatEvent) {
         val message = event.message
-        if (message.contains("Starting in 1 second.") && IslandType.CATACOMBS.onIsland()) {
-            DungeonStartEvent().post()
+        if (message.contains("Starting in 1 second.") && IslandType.CATACOMBS.onIsland() && lastDunegonStartEventSend + 3000 < time) {
+            DungeonStartEvent().post().also { if (!it) lastDunegonStartEventSend = time }
```

Comment:
```suggestion
            DungeonStartEvent().post()
            lastDunegonStartEventSend = time
```

dungeon start/end event should never get canceled anyway so we dont need to check if it got cancelled